registrations/models.py

```diff
@@ -52,25 +78,33 @@ class Registration(models.Model):
     enrolment_start_time = models.DateTimeField(
         verbose_name=_("Enrollment start time"), blank=True, null=True
     )
+
     enrolment_end_time = models.DateTimeField(
         verbose_name=_("Enrollment end time"), blank=True, null=True
     )
 
     confirmation_message = models.TextField(
         verbose_name=_("Confirmation message"), blank=True, null=True
     )
+
     instructions = models.TextField(
         verbose_name=_("Instructions"), blank=True, null=True
     )
 
     maximum_attendee_capacity = models.PositiveSmallIntegerField(
         verbose_name=_("Maximum attendee capacity"), null=True, blank=True
     )
+
     minimum_attendee_capacity = models.PositiveSmallIntegerField(
         verbose_name=_("Minimum attendee capacity"), null=True, blank=True
     )
+
     waiting_list_capacity = models.PositiveSmallIntegerField(
-        verbose_name=_("Minimum attendee capacity"), null=True, blank=True
+        verbose_name=_("Waiting list capacity"), null=True, blank=True
+    )
+
+    mandatory_fields = models.ManyToManyField(
```

Comment:
Something that I was thinking about was that do we need to have `MandatoryField` as a model to enable us to define new fields dynamically? To me it seems that mostly we're trying to list what fields from `SignUp` and mandatory for this registration instance. Or do we need to be able to define a field `foobar` or some other field that doesn't exist on `SignUp` model? `DefaultMandatoryField` seems to resemble actual fields on `SignUp` model.

How would you feel about something like this? (Note, this is a dry-coded example)

```python
from django.contrib.postgres.fields import ArrayField

class MandatoryFields(models.TextChoices):
    """Choices for mandatory fields on SignUp model."""
    CITY = "city", _("City")
    NAME = "name", _("Name")
    PHONE_NUMBER = "phone_number", _("Phone number")
    STREET_ADDRESS = "street_address",  _("Street address")
    ZIPCODE = "zipcode", _("ZIP code")


class Registration(models.Model):

    mandatory_fields = ArrayField(
        models.CharField(
            max_length=16,
            choices=MandatoryFields.choices,
            blank=True,
        ),
        default=list,
    )
```

Reply:
This is really good idea and makes also front-end implementation much simpler. Updated the PR and added mandatory fields to registration form in Django admin as well.